sockeye/utils.py

```diff
@@ -774,16 +787,15 @@ def log_parameters(params: mx.gluon.ParameterDict):
     learned_parameter_names = []
     #info = []  # type: List[str]
     for name, param in sorted(params.items()):
-        repr = "%s [%s, %s]" % (name, param.shape, param.dtype)
-        #info.append("%s shape=%s, dtype=%s" % (name, param.shape, param.dtype))
+        repr = "%s [%s, %s]" % (name, param.shape, _print_dtype(param.dtype))
         if param.grad_req == 'null':
             fixed_parameter_names.append(repr)
         else:
             learned_parameter_names.append(repr)
     #percent_fixed = 100 * (fixed_parameters / max(1, total_parameters))
     #percent_learned = 100 * (learned_parameters / max(1, total_parameters))
-    logger.info("Trainable parameters: %s", ", ".join(learned_parameter_names))
-    logger.info("Fixed model parameters: %s", ", ".join(fixed_parameter_names))
+    logger.info("Trainable parameters:\n%s", pprint.pformat(learned_parameter_names))
+    logger.info("Fixed model parameters:\n%s", pprint.pformat(fixed_parameter_names))
     #logger.info("Fixing %d parameters (%0.2f%%)", fixed_parameters, percent_fixed)
     #logger.info("Learning %d parameters (%0.2f%%)", learned_parameters, percent_learned)
     #logger.info("Total # of parameters: %d", total_parameters)
```

Comment:
Great update!  While we're doing work here, can we reenable printing these parameter totals?

Reply:
I'd love to, the problem is deferred initialization of some parameters so that the shape of these is still `0` in some dimensions at this point. Their actual size can only be known after a first batch of data has been passed to the model. Doing this here would be pretty ugly. Do you have a better idea?

Reply:
That's a good reason not to report parameters here.  We could move the report to the first batch or checkpoint with a label the user could grep for.  That's probably out of scope for this commit.